libpod/container_inspect.go

```diff
@@ -743,22 +743,21 @@ func (c *Container) generateInspectContainerHostConfig(ctrSpec *spec.Spec, named
 	ipcMode := ""
 	if c.config.IPCNsCtr != "" {
 		ipcMode = fmt.Sprintf("container:%s", c.config.IPCNsCtr)
-	} else {
+	} else if ctrSpec.Linux != nil {
 		// Locate the spec's IPC namespace.
 		// If there is none, it's ipc=host.
 		// If there is one and it has a path, it's "ns:".
 		// If no path, it's default - the empty string.
-		foundIPCNS := false
+
 		for _, ns := range ctrSpec.Linux.Namespaces {
 			if ns.Type == spec.IPCNamespace {
-				foundIPCNS = true
 				if ns.Path != "" {
 					ipcMode = fmt.Sprintf("ns:%s", ns.Path)
 				}
 				break
 			}
 		}
-		if !foundIPCNS {
+		if ipcMode == "" {
```

Comment:
Is this actually equivalent? I'm trying to work my head around the logic here, and I think it may be setting host when we're actually private?

Reply:
Good point, you actually discovered a bug.  If we have a namespace then it should be "private" else it is host.  So in the loop above we should set ipdMode = "private" if there is no path.